fluent-plugin-carbon-v2/lib/fluent/plugin/filter_carbon_v2.rb

```diff
@@ -1,13 +1,161 @@
-
-
-require "fluent/plugin/filter"
+require 'fluent/plugin/filter'
 
 module Fluent
   module Plugin
+    # fluentd plugin for convert data point json to carbon v2
     class CarbonV2Filter < Fluent::Plugin::Filter
-      Fluent::Plugin.register_filter("carbon_v2", self)
+      Fluent::Plugin.register_filter('carbon_v2', self)
+
+      helpers :record_accessor
+
+      config_param(
+        :relabel,
+        :hash,
+        default: {},
+        symbolize_keys: false,
+        value_type: :string
+      )
+      config_param(
+        :inclusions,
+        :hash,
+        default: {},
+        symbolize_keys: false,
+        value_type: :string
+      )
+      config_param(
+        :exclusions,
+        :hash,
+        default: {},
+        symbolize_keys: false,
+        value_type: :string
+      )
+      config_param :strict_inclusions, :bool, default: false
+      config_param :strict_exclusions, :bool, default: false
+      config_param :sort_labels, :bool, default: true
+
+      def configure(conf)
+        super
+        @inclusions.each do |key, value|
+          @inclusions[key] = Regexp.new(value)
+        end
+        @exclusions.each do |key, value|
+          @exclusions[key] = Regexp.new(value)
+        end
+        @metric_accessor = record_accessor_create("$.#{KEY_METRIC}")
+        @timestamp_accessor = record_accessor_create("$.#{KEY_TIMESTAMP}")
+        @value_accessor = record_accessor_create("$.#{KEY_VALUE}")
+      end
+
+      def multi_workers_ready?
+        true
+      end
+
+      def filter(_tag, _time, record)
+        dotified = dotify_record(record)
+        relabeled = relabel_record(dotified)
+        wrap(to_carbon_line(relabeled)) if valid?(relabeled)
+      end
+
+      private
+
+      KEY_METRIC = '@metric'.freeze
+      KEY_TIMESTAMP = '@timestamp'.freeze
+      KEY_VALUE = '@value'.freeze
+      KEY_MESSAGE = 'message'.freeze
+      SPLITOR = '.'.freeze
```

Comment:
sp: SPLITTER

Reply:
Not sure. In my mind these two words are different:

https://github.com/search?q=splitor&type=Code

https://github.com/search?q=splitter&type=Code

The earlier one may be more suitable in this case?

Reply:
Ahh, my bad. I did a quick google search but didn't find anything so I assumed you meant splitter :(